### DISCUSSION TITLE
Condition si - Cases à cocher

### DISCUSSION THREAD

**Discussion by JolyaneL** [MTESSDev/FRW#297] [Annonces]
Bonjour, 

Je souhaite mettre à la négative une condition dans une question comprenant six cases à cocher.
`v-if: !val('maQuestion').includes('opt6')`

Au lieu de faire la condition complète inverse,
`v-if: val('maQuestion').includes('opt1') || val('maQuestion').includes('opt2') || val('maQuestion').includes('opt3') || val('maQuestion').includes('opt4') || val('maQuestion').includes('opt5')`

J'ai vu dans la documentation que c'est inclut dans un exemple, mais de mon côté ça bloque la prévisualisation.
![image](https://github.com/user-attachments/assets/14dfc522-6422-4fb5-a683-ee1983179f3a)

**Comment by Hutson48** [comment]
Bonjour,

Merci pour votre question.

Concernant l'utilisation de la condition pour exclure une case à cocher, il est effectivement possible d'inverser la logique en utilisant le point d'exclamation pour exclure une valeur spécifique. Toutefois, si cela bloque la prévisualisation, il pourrait s'agir d'un problème lié à la manière dont le formulaire interprète les conditions ou à une incompatibilité dans la logique appliquée. 

YAML est très sensible à la mise en forme et aux caractères spéciaux. Par exemple, l’utilisation du point d’exclamation (!) doit être gérée avec précaution, car il est aussi utilisé dans YAML comme opérateur de type spécial. Essayez de l'encadrer de guillemets pour éviter toute confusion: 
````yaml
v-if: "!val('maQuestion').includes('opt6')"
````
Si le problème persiste, de m’envoyer plus de détails sur votre configuration actuelle afin que je puisse vous répondre.

Merci et bonne journée,

**Reply by JolyaneL** [reply]
Effectivement, cela fonctionne parfaitement un grand merci.
